tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -365,6 +365,86 @@ Type ConvertType(Type t) {
   return t;
 }
 
+Value GetUniqueResultAlloc(
+    Value oper, Location loc,
+    ConversionPatternRewriter& rewriter,
+    SmallVector<Value, 4> alloc_operands) {
+  auto output_tensor_type = oper.getType().dyn_cast<RankedTensorType>();
+  if (!output_tensor_type) {
+    emitError(loc, "tensor to buffer conversion expects valid result type");
+    return nullptr;
+  }
+  if (output_tensor_type.getShape().size() != 1) {
+    emitError(loc, "UniqueOp only suport 1-D ranked tensor.");
+    return nullptr;
+  }
+
+  auto output_tensor_memref_type =
+      MemRefType::get(output_tensor_type.getShape(), output_tensor_type.getElementType());
+  Operation* def_op = oper.getDefiningOp();
+  auto block = def_op->getBlock();
+  OpBuilder alloc_builder(def_op);
+  auto alloc = alloc_builder.create<AllocOp>(
+      loc, output_tensor_memref_type, alloc_operands);
+  alloc.setAttr(kTempBufferAttr, rewriter.getBoolAttr(true));
+  alloc_builder.setInsertionPoint(block, std::prev(block->end()));
+  alloc_builder.create<DeallocOp>(loc, alloc);
+
+  return alloc;
+}
+
+struct HloToLHloUniqueOpConverter
+    : public OpConversionPattern<xla_hlo::UniqueOp> {
+ public:
+  using OpConversionPattern::OpConversionPattern;
+
+  PatternMatchResult matchAndRewrite(
+    xla_hlo::UniqueOp op, ArrayRef<Value> operands,
+    ConversionPatternRewriter& rewriter) const final {
+    auto loc = op.getLoc();
+    // Unique only support 1-D tensor
+    auto input_tensor_type = op.x().getType().dyn_cast<RankedTensorType>();
+    if (!input_tensor_type) {
+      emitError(op.getLoc(), "Only suport ranked tensor now.");
```

Comment:
Don't emit errors for mismatches, there could be multiple passes and only one that matches. Same below.